Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/ForegroundRiskCalculationService.kt

```diff
@@ -35,13 +47,24 @@ class WatchdogService @Inject constructor(
         context.applicationContext.getSystemService(Context.WIFI_SERVICE) as WifiManager
     }
 
+    private val mutex = Mutex()
+
+    private var latestRunTimeStamp: Instant = Instant.EPOCH
+
     override fun initialize() {
-        val isAutoModeEnable = runBlocking { backgroundModeStatus.isAutoModeEnabled.first() }
-        // Only do this if the background jobs are enabled
-        if (!isAutoModeEnable) {
-            Timber.tag(TAG).d("Background jobs are not enabled, aborting.")
-            return
+        foregroundState.isInForeground
+            .distinctUntilChanged()
+            .filter { it }
+            .onEach { runRiskCalculations() }
+            .launchIn(appScope)
+    }
+
+    private suspend fun runRiskCalculations() = mutex.withLock {
+        if (Duration.between(latestRunTimeStamp, timeStamper.nowJavaUTC) < minTimeBetweenRuns) {
```

Comment:
1- this could problematic in time travel cases, maybe you consider calculating `abs`tract value of time difference ?
2- As you already mentioned above `Task` already taking care of aborting if redundant calculation is requested , is that still needed here ?

Reply:
yes, it will abort but the UI would show the in progress state each time the user opens the home screen which is distraction